plugin-code/src/main/java/com/trivago/cluecumber/json/postprocessors/ElementPostProcessor.java

```diff
@@ -87,56 +87,30 @@ private void processAttachments(final List<Step> steps, List<ResultMatch> afterH
         }
     }
 
-    /**
-     * Save images to files, clear their Base64 content from JSON and store their filenames in order to save memory
+   /**
+     * Save embeddings to files, clear their Base64 content from JSON and store their filenames in order to save memory
      *
      * @param embeddings The {@link Embedding} list.
      */
     private void processEmbedding(final List<Embedding> embeddings) {
         for (Embedding embedding : embeddings) {
-            if (embedding.isImage()) {
-                String filename = saveImageEmbeddingToFileAndGetFilename(embedding);
-                embedding.setFilename(filename);
-            }
+        	String filename = saveEmbeddingToFileAndGetFilename(embedding);
+            embedding.setFilename(filename);
             attachmentIndex++;
         }
     }
 
     /**
-     * Saves image attachments to a file and returns the filename.
+     * Saves attachments to a file and returns the filename.
      *
      * @param embedding The {@link Embedding} to process.
      * @return The filename to the processed image.
      */
-    private String saveImageEmbeddingToFileAndGetFilename(final Embedding embedding) {
-        if (!embedding.isImage()) {
-            return "";
-        }
-
-        String fileEnding;
-        switch (embedding.getMimeType()) {
-            case "image/png":
-                fileEnding = ".png";
-                break;
-            case "image/jpeg":
-                fileEnding = ".jpg";
-                break;
-            case "image/gif":
-                fileEnding = ".gif";
-                break;
-            case "image/svg+xml":
-            case "image/svg":
-                fileEnding = ".svg";
-                break;
-            default:
-                fileEnding = ".unknown";
-        }
-
-        byte[] dataBytes = Base64.decodeBase64(embedding.getData().getBytes(StandardCharsets.UTF_8));
-
+    private String saveEmbeddingToFileAndGetFilename(final Embedding embedding) {        
         // Clear attachment data to reduce memory
-        embedding.setData("");
-
+        // embedding.setData("");   // Can not set this to empty string. This is now required in getDecodedData() function in Embedding.java
```

Comment:
Removing this line causes all encoded attachments (including large Base64 encoded image strings) to remain in memory. This should be reverted and handled differently.

In general, instead of commenting out code, it should be removed instead.

Reply:
Yes, this takes lots of memory. But If we set "embedding.setData("");" after saving the attachments, it works only for image attachments. In order to render other formats attachments, we must have the encoded data for decoding purpose in memory.

Or we may have to use the saved attachments again. I think, For this to work, we have to again somehow encode and decode attachments for rendering. Please suggest your thoughts on this.

Reply:
For the non-image cases, we could either decode the data and write it back to `data` or introduce a new property to store the decoded data.

Reply:
new property to store the decoded data.